notify-debouncer-mini/src/lib.rs

```diff
@@ -160,88 +158,126 @@ pub struct DebouncedEvent {
 }
 
 impl DebouncedEvent {
+    #[inline(always)]
     fn new(path: PathBuf, kind: DebouncedEventKind) -> Self {
         Self { path, kind }
     }
 }
 
-type DebounceData = Arc<Mutex<DebounceDataInner>>;
+enum InnerEvent {
+    NotifyEvent(Result<Event, Error>),
+    Shutdown,
+}
 
-#[derive(Default)]
 struct DebounceDataInner {
+    /// Path -> Event data
     d: HashMap<PathBuf, EventData>,
+    /// timeout used to compare all events against, config
```

Comment:
I always questioned what is the motivation to remove any information about the events? For example we in alacritty don't want to handle `Removed` events at all to not try to load from the removed path. But that way when you map everything to any event the information for us is sort of discarded. Might forward the events untouched all-together? Will simplify the logic and you can just simply `move` the vector into users.

Reply:
If you want specific events, use notify. If you only need information that something changed, use the mini-debouncer. Which is what most people do: Recompute things (assets, binaries,..) when the files change. Debouncing with all event details won't make much sense, you'll just receive a Vec of changes for each path and then have to look through that in order, while the data you got from debouncing might already be wrong the moment you receive them, so concluding anything based on the Vec is wrong. Flattening events to one (create + delete) is something we did, but removed for various problems, as documented in the upgrading guide. Moreover you can't rely on events to not miss out on events, thus you shouldn't ever rely on events for the actual state of files #412

Reply:
Hm, but at this point, do we really need timestamps from `EventData`? What is the use case for them? Could just return `HashSet<PathBuf>` at this point?

Reply:
We don't return EventData, that is just for internal tracking. Otherwise we would report events even if they haven't timed out, so you may be trying to read the config while it's still changing. But if you want to, you can still handle `AnyContinuous`, in case you want to handle the fact something is written for longer than the timeout.